src/vs/workbench/api/common/extHostLanguageFeatures.ts

```diff
@@ -1062,6 +1062,26 @@ class SignatureHelpAdapter {
 	}
 }
 
+class InlineHintsAdapter {
+	private readonly _cache = new Cache<vscode.InlineHint[]>('InlineHints');
```

Comment:
Caching is only required when having a `resolve`-function on a provider. Right now this is a leak